plugins/mapper-annotated-text/src/main/java/org/elasticsearch/index/mapper/annotatedtext/AnnotatedTextFieldMapper.java

```diff
@@ -229,28 +209,6 @@ public boolean intersects(int start, int end) {
                     || (start <= endOffset && end >= endOffset)
                     || (start >= offset && end <= endOffset);
             }
-
-            @Override
-            public int hashCode() {
```

Comment:
This class has a very specific hashCode, is there perhaps a performance implication if we change how we compute the hash?

Reply:
It seems to be a standard boilerplate hash code generation. Records use MethodHandles to generate a `hashCode` implementation  which should provide similar performance to a hand-written one.  For example, some benchmarks here: https://github.com/wyhasany/lombokVsRecord